backend/src/main/java/com/huellapositiva/domain/actions/UpdateVolunteerProfileAction.java

```diff
@@ -0,0 +1,140 @@
+package com.huellapositiva.domain.actions;
+
+import com.huellapositiva.domain.exception.MatchingEmailException;
+import com.huellapositiva.domain.model.valueobjects.EmailAddress;
+import com.huellapositiva.domain.model.valueobjects.Id;
+import com.huellapositiva.application.dto.ProfileDto;
+import com.huellapositiva.domain.service.EmailCommunicationService;
+import com.huellapositiva.infrastructure.orm.entities.JpaLocation;
+import com.huellapositiva.infrastructure.orm.entities.JpaVolunteer;
+import com.huellapositiva.infrastructure.orm.repository.JpaCredentialRepository;
+import com.huellapositiva.infrastructure.orm.repository.JpaVolunteerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.time.LocalDate;
+
+@Service
+public class UpdateVolunteerProfileAction {
+    @Autowired
+    private JpaVolunteerRepository jpaVolunteerRepository;
+
+    @Autowired
+    private JpaCredentialRepository jpaCredentialRepository;
+
+    @Autowired
+    private EmailCommunicationService emailCommunicationService;
+
+    /**
+     * This method update the user profile information in database
+     *
+     * @param profileDto New user profile information to update
+     * @param email Email of user logged
+     * @throws IOException
+     */
+    public void execute(ProfileDto profileDto, String email) throws IOException {
+        if (someFieldIsEmptyCredentials(profileDto) || someFieldEmptyLocation(profileDto)) {
+            throw new IOException("Some field is null");
+        }
+
+        boolean isNotEqualsEmail = !email.equals(profileDto.getEmail());
+        if (isNotEqualsEmail && jpaCredentialRepository.findByEmail(profileDto.getEmail()).isPresent()) {
+            throw new MatchingEmailException("Email already exists in the database.");
+        }
+
+        JpaLocation jpaLocation = updateLocation(profileDto, email);
+        JpaVolunteer jpaVolunteer = jpaVolunteerRepository.findByEmailProfileInformation(email);
+        updateProfileInformation(profileDto, jpaVolunteer);
+        updateCredentials(profileDto, jpaVolunteer);
+
+        jpaVolunteer.setLocation(jpaLocation);
+        jpaVolunteerRepository.save(jpaVolunteer);
+
+        if(isNotEqualsEmail){
+            emailCommunicationService.sendMessageEmailChanged(EmailAddress.from(email));
```

Comment:
Missing changing role to XXX_NOT_CONFIRMED.

Test after changing email you can still login with your new role
Test that if email cannot be sent the profile is still updated

Reply:
It's done.